### ISSUE TITLE
Resolving wsl+Ubuntu server installation is incorrect

### ISSUE THREAD

**Issue by holofermes** [microsoft/vscode-remote-release#4311]
Hey team,

I am running into an issue when opening a local WSL1 remote  connection, and although I think I know why this is happening, I'd like some guidance on how to best proceed here.

These are my steps:

- From a vscode running on a mac I connected to a windows box, which had no vscode installed.
- I changed my WSL1 $HOME path to be the same as my windows path, so instead of being `/home/meuser/` is `/mnt/c/Users/meuser/`
- I installed vscode and the remote extension on the windows box, and tried to connect to WSL1

Looking at the full log below, it seems that the WSL extension thinks to have found the server installation, when in reality that is the installation that I triggered when I first connected to the windows box when vscode wasn't even installed.

```
[2021-01-17 04:53:22.292] Resolving wsl+Ubuntu, resolveAttempt: 1
[2021-01-17 04:53:22.339] Starting VS Code Server inside WSL (Ubuntu)
[2021-01-17 04:53:22.339] Extension version: 0.52.0, Windows build: 18363. Multi distro support: available. WSL path support: enabled
[2021-01-17 04:53:22.340] No shell environment set or found for current distro.
[2021-01-17 04:53:22.450] Probing if server is already installed: C:\WINDOWS\System32\wsl.exe -d Ubuntu -e sh -c "[ -d ~/.vscode-server/bin/hash123abc ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
[2021-01-17 04:53:22.551] Probing result: found
[2021-01-17 04:53:22.551] Server install found in WSL
[2021-01-17 04:53:22.552] Launching C:\WINDOWS\System32\wsl.exe -d Ubuntu sh -c '"$VSCODE_WSL_EXT_LOCATION/scripts/wslServer.sh" hash123abc stable .vscode-server 0  --disable-telemetry'}
[2021-01-17 04:53:22.671] Setting up server environment: Looking for /mnt/c/Users/meuser/.vscode-server/server-env-setup. Not found.
[2021-01-17 04:53:22.671] WSL version: 4.4.0-18362-Microsoft Ubuntu
[2021-01-17 04:53:22.671] Starting server: /mnt/c/Users/meuser/.vscode-server/bin/hash123abc/server.sh  --port=0 --use-host-proxy --fileWatcherPolling=0 --disable-telemetry
[2021-01-17 04:53:22.671] /mnt/c/Users/meuser/.vscode-server/bin/hash123abc/server.sh: 12: /mnt/c/Users/meuser/.vscode-server/bin/hash123abc/server.sh: /mnt/c/Users/meuser/.vscode-server/bin/hash123abc/node: not found
[2021-01-17 04:53:22.672] VS Code Server for WSL closed unexpectedly.
[2021-01-17 04:53:22.672] For help with startup problems, go to
[2021-01-17 04:53:22.672] https://code.visualstudio.com/docs/remote/troubleshooting#_wsl-tips
[2021-01-17 04:53:22.679] WSL Daemon exited with code 0
 ```

**Comment by bamurtaugh** [comment]
Hi @holofermes, thanks for writing this up. Just to make sure I understand what you're saying:

- You use Remote-SSH to connect a Mac to Windows 
- Change WSL1 $Home path on remote Windows machine (did you change this while connected via Remote-SSH?)
- You installed VS Code and Remote - WSL on the Windows machine. How did you try to connect to a remote WSL host? Could you give some more details on the steps you took? Was it anything like the following? https://www.hanselman.com/blog/the-easy-way-how-to-ssh-into-bash-and-wsl2-on-windows-10-from-an-external-machine

**Comment by PavelSosin-320** [comment]
@holofermes It is impossible to connect to WSL VM from outside of Windows machine. By the intention! This is WSL networking architecture. WSL has no IP known outside Windows, no bridge. Do you want to use your OEM Windows Home edition with enabled WSL as a server?

**Comment by holofermes** [comment]
Hey all, I ended up working around the problem but I want to make sure I outline a little more clearly the series of events that led to the issue:

On the WINDOWS side I have a windows 10 Home with a single user `userme`, without vscode installed, and wsl1 with a user with the same name.

1 WINDOWS: Enabled OpenSSH from the [settings UI](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse#installing-openssh-from-the-settings-ui-on-windows-server-2019-or-windows-10-1809)
2 MAC: Connected from to the windows box using vscode-remote
- WINDOWS: The `vscode-server` folder now exists here: `C:\Users\meuser\.vscode-server` because of the default OpenSSH configuration

3 WINDOWS: Changed wsl1 user home from `/home/userme` to `/mnt/c/Users/meuser` (which is `C:\Users\meuser`)
4 WINDOWS: Installed vscode, attemp to connection with remote-wsl against the local wsl1
5 Error

Step 3 is what enables the problem, and since I don't need this my solution was just undoing this step, and letting wsl1 default home folder.

Thanks!
